### ISSUE TITLE
Error after apply - Stored plan does not exist 

### ISSUE THREAD

**Issue by chvima** [diggerhq/digger#1110]
I have started to use the config: upload-plan-destination: github but cannot do digger apply successfully because .
Using v0.3.19 of both self hosted backend and github action.

Logs after digger plan:
```
With the provided path, there will be 1 file uploaded
Artifact name is valid!
Root directory input is valid!
Beginning upload of artifact content to blob storage
Uploaded bytes 24636
Finished uploading artifact content to blob storage!
SHA256 hash of uploaded artifact zip is 8c1fc31fcdd9d33710de40fa466400756417b2672ec1e9fc898b56fe1f14e9d3
Finalizing artifact upload
Artifact plans-.zip successfully finalized. Artifact ID 1215300612
Artifact plans- has been successfully uploaded! Final size is 24636 bytes. Artifact ID is 1215300612
Artifact download URL: https://github.com/xxx/yyy/actions/runs/7756839457/artifacts/1215300612
```

I can see the artifact is present in the action run Summary:
![image](https://github.com/diggerhq/digger/assets/104632791/649357f8-5356-4251-a99e-72a408ad5976)

But after digger apply I get an error:
```
13:50:57 digger.go:356: PR status, mergeable: true, merged: false
13:50:57 digger.go:371: Failed to retrieve stored plan. stored plan does not exist
13:50:58 main.go:938: Failed to run commands. error while running command: Failed to retrieve stored plan. stored plan does not exist
```

Do you have an idea where the bug could be or whether it is something in my config? 
Or could you add more debug messages into that function to be able to see what package from where does it try to fetch?

**Comment by ZIJ** [comment]
Thanks for reporting @chvima! I'm taking a look; at the first glance it seems that the artifact is misnamed for some reason; normally it's named "plans-{{prNumber}}" e.g. "plans-11.zip". So it appears that the correct name is not being set for some reason.

We made some changes affecting plan storage in v0.3.19; could you try v0.3.18 also, whether or not it behaves the same? That'd help us to isolate the issue faster

**Comment by chvima** [comment]
Alright, I will try on monday and let you know. 
Also I don’t fully understand the design of plan storage in Github - those are now stored in specific plan workflow run. Each digger plan stores it’s own artifact in it’s workflow run. What logic does the flow for digger apply use to retrieve those - the last run of plan action?

**Comment by chvima** [comment]
@ZIJ With v0.3.18 I have same issue, same name of the artifact. Same with v0.3.20

**Comment by chvima** [comment]
I think the problem with wrong plan name might be here:
https://github.com/diggerhq/digger/blob/develop/action.yml#L311

github.event.issue.number
github.event.number

In the github context of my digger runs there is neither of those, since the Github workflow is run by workflow_dispatch by the Github Application (not on: pull_request or on: issue_comment)

**Comment by ZIJ** [comment]
Observing that on the screenshot there is "plans-" without a PR number. Which makes it related to #1149

**Comment by motatoes** [comment]
> I think the problem with wrong plan name might be here:
https://github.com/diggerhq/digger/blob/develop/action.yml#L311

> github.event.issue.number
> github.event.number

> In the github context of my digger runs there is neither of those, since the Github workflow is run by workflow_dispatch by > the Github Application (not on: pull_request or on: issue_comment)

I think this is exactly the cause @chvima . I wonder if we can just upload from digger without relying on subsequent steps. My teammate mentioned that github doesn't expose the artefact upload API publically except through the action `actions/upload-artifact `. 

Need to figure out a way to either upload directly from digger or pass the PR number from orchestrator to the job

**Comment by chvima** [comment]
Closing, after update to 0.4.6 (last version I had was 0.3.25) it is working, has been fix by one of the changes in 0.4.X
